python/interpret_community/explanation/explanation.py

```diff
@@ -1148,25 +1148,41 @@ def eval_data(self):
         :return: The dataset or dataset ID.
         :rtype: list or str
         """
-        return self._eval_data
+        return self._convert_to_list(self._eval_data)
 
     @property
     def eval_y_predicted(self):
         """Get predicted ys for the evaluation data.
 
         :return: The predicted ys for the evaluation data.
-        :rtype: np.array
+        :rtype: list
         """
-        return self._eval_y_predicted
+        return self._convert_to_list(self._eval_y_predicted)
 
     @property
     def eval_y_predicted_proba(self):
         """Get predicted probability ys for the evaluation data.
 
-        :param eval_ys_predicted_proba: The predicted probability ys for the evaluation data.
-        :type eval_ys_predicted_proba: np.array
+        :return: The predicted probability ys for the evaluation data.
+        :rtype: list
         """
-        return self._eval_y_predicted_proba
+        return self._convert_to_list(self._eval_y_predicted_proba)
+
+    def _convert_to_list(self, data):
+        """Convert data to a Python list.
+
+        :param data: The data to be converted.
+        :type data: np.array, pd.DataFrame, list, scipy.sparse
+        :return: The data converted to a list (except for sparse which is unchanged).
+        :rtype: list | scipy.sparse
```

Comment:
for multiclass case:
| list[scipy.sparse]

Reply:
will update